### PR TITLE
SW-3127 Remove references to species.endangered

### PR DESCRIPTION
**Pull request by sgrimm**
Stop reading and writing the `endangered` column on the species table, in
preparation for dropping it.

The species API payloads still have `endangered` fields, but they are now
calculated from the conservation category.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1255** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1255" target="_blank">Graphite</a> 
    * **PR #1256** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1256" target="_blank">Graphite</a> 
      * **PR #1257** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1257" target="_blank">Graphite</a> 
        * **PR #1258** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1258" target="_blank">Graphite</a>  👈
          * **PR #1259** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1259" target="_blank">Graphite</a> 
            * **PR #1260** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1260" target="_blank">Graphite</a> 
              * **PR #1261** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1261" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1258?utm_source=stack-comment).

File: src/main/kotlin/com/terraformation/backend/search/table/SpeciesTable.kt [hunk 1]
```diff
@@ -42,7 +44,11 @@ class SpeciesTable(tables: SearchTables) : SearchTable() {
           timestampField("checkedTime", SPECIES.CHECKED_TIME),
           textField("commonName", SPECIES.COMMON_NAME),
           enumField("conservationCategory", SPECIES.CONSERVATION_CATEGORY_ID, localize = false),
-          booleanField("endangered", SPECIES.ENDANGERED),
+          booleanField(
+              "endangered",
+              DSL.case_(SPECIES.CONSERVATION_CATEGORY_ID)
+                  .`when`(ConservationCategory.Endangered, true)
```
Comment (charlie-tf) [inline src/main/kotlin/com/terraformation/backend/search/table/SpeciesTable.kt:50]: Does this include the whole series of categories of Vulnerable or worse, or only Endangered?

File: src/main/kotlin/com/terraformation/backend/species/api/SpeciesController.kt [hunk 2]
```diff
@@ -234,7 +234,7 @@ data class SpeciesResponseElement(
       ecosystemTypes = model.ecosystemTypes.ifEmpty { null },
       commonName = model.commonName,
       conservationCategory = model.conservationCategory,
-      endangered = model.endangered,
+      endangered = model.conservationCategory == ConservationCategory.Endangered,
```
Comment (charlie-tf) [inline src/main/kotlin/com/terraformation/backend/species/api/SpeciesController.kt:237]: Same question here. Looking like here it's only Endangered?

File: src/main/kotlin/com/terraformation/backend/search/table/SpeciesTable.kt (hunk 1 above)
Comment (sgrimm) [reply src/main/kotlin/com/terraformation/backend/search/table/SpeciesTable.kt:50]: Only Endangered; that was the request from the product team (see discussion on SW-3127).

This version of the code shouldn't be active for too long; I think the plan would be to roll out this stack of changes and the corresponding terraware-web change in rapid succession.

Comment (charlie-tf) [reply src/main/kotlin/com/terraformation/backend/search/table/SpeciesTable.kt:50]: Got it, thanks for linking the discussion

**Pull request merged**